atom/browser/browser_mac.mm

```diff
@@ -60,6 +60,25 @@
   return return_code == noErr;
 }
 
+bool Browser::IsDefaultProtocolClient(const std::string& protocol) {
+  if (protocol.empty())
+    return false;
+
+  NSString* identifier = [base::mac::MainBundle() bundleIdentifier];
+  if (!identifier)
+    return false;
+
+  NSString* protocol_ns = [NSString stringWithUTF8String:protocol.c_str()];
+  
+  CFStringRef bundle =
+      LSCopyDefaultHandlerForURLScheme(base::mac::NSToCFCast(protocol_ns));
+  if (!bundle){
+    return false;
+  }
+  NSString* bundleId = base::mac::CFToNSCast(bundle);
```

Comment:
The `bundle` is a `CFStringRef` so it has to be released, you can reference how Chromium did it:
https://code.google.com/p/chromium/codesearch#chromium/src/chrome/browser/shell_integration_mac.mm&sq=package:chromium&type=cs&l=34